### DISCUSSION TITLE
User-Agent kullanmak

### DISCUSSION THREAD

**Discussion by Parmendies** [19/classroom#75] [Anonim] [answered]
Merhabalar hocam. Google lens ile çeviriye benzer bir uygulama yapıyorum. Api için TKD'nın end pointini kullanıcam. [https://sozluk.gov.tr/gts?ara=merhaba](url) Her hangi bi api doc bulamadım ve konvansiyon da yok.  Ayrıca uygulamadan istek atınca  "Connection closed before full header was received" hatası alıyorum. Sanki bütün bunlar bana bir şey anlatıyor gibi ama :) yine de uygulamayı TDK ile bağlamak çok istiyorum. Bu şekilde bir kullanım sorun oluşturur mu ?
```dart
    http.Response response;
    // const String baseUrl = 'https://sozluk.gov.tr/gts';
    const String userAgent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3';

    final Uri uri = Uri(
      scheme: 'https',
      host: 'sozluk.gov.tr',
      path: 'gts',
      queryParameters: {'ara': word},
    );

    try {
      response = await http.get(
        uri,
        headers: {'User-Agent': userAgent},
      );
    } catch (e) {
      return null;
    }
```

**Comment by roktas** [comment, marked as answer]
Gördüğüm kadarıyla dokümante edilmemiş basit bir API sunuyorlar. TDK böyle bir kamu servisini niye daha düzgün sunmaz bilmiyorum. Dokümantasyonu bir yana authz, trafik kısıtlaması vs olmasını beklerdim. User-Agent set edebilirsin. Örneğin komut satırında `curl -fsSL --verbose 'https://sozluk.gov.tr/gts?ara=merhaba'` ile denediğimde başarıyla sonuç alıyorum (curl User-Agent'i `curl/7.88.1` olarak ayarlıyor, servis JSON dönüyor).

**Reply by Parmendies** [reply]
Teşekkür ederim hocam.
